src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -442,6 +494,12 @@ private BoundStatement BindGoto(GotoStatementSyntax node, DiagnosticBag diagnost
 
         private BoundStatement BindLocalFunctionStatement(LocalFunctionStatementSyntax node, DiagnosticBag diagnostics)
         {
+            if (IsEmbeddedStatementParent(node.Parent))
```

Comment:
>if (IsEmbeddedStatementParent(node.Parent)) [](start = 12, length = 43)

It feels like it would be better to intercept this statement in BindPossibleEmbeddedStatement function instead.